handle unknown index types gracefully in iUtils

alperkokmen opened the pull request:
this updates `parseTaskId` in `iUtils.coffee` which uses a somwhat strict regular expression to parse a given task id and extract `type`, `datasource`, and time (relates to #3 and #9).

this simply removes throwing the error and falls back to `'other'` for type leaving `dataSource` and `dataTime` alone. in case of a match failure, they would just default to undefined which isn't a blocker to render `/indexing-service` view.

alperkokmen committed:
handle unknown index types gracefully in iUtils

this updates parseTaskId in iUtils.coffee which uses a somwhat strict
regular expression to parse a given task id and extract type,
datasource, and time.

this simply removes throwing the error and falls back to 'other' for
type leaving dataSource and dataTime alone. in case of a match failure,
they would just default to undefined which isn't a blocker to render
/indexing-service view.

alperkokmen commented:
hey @xvrl, could you please take a look when you get a chance?

xvrl commented:
Hi @alperkokmen, I'm not sure I'm the best person to review this. @drcrallen looks like you've made changes to this logic before, mind taking a look?

alperkokmen commented:
just came across a new issue (https://github.com/druid-io/druid-console/issues/19) that these changes would take care of. @drcrallen, could you please take a look?

alperkokmen commented:
bump – @drcrallen, could you please take a look or tag someone who can review/merge this change? thanks!

clintropolis reviewed (changes requested):
Could you add a unit test for this behavior?

alperkokmen committed:
fix regression in parseTaskId regex

when 1b74f29 added support to handle archive tasks, it introduced a new
matching group for the regular expression in parseTaskId which shifted
the indexing for resulting matches (i.e. m).

to simplify things, i am making marking the outer group as
don't-capture-just-group (?:) to keep the matching result consistent. as
a result, type will be populated using m[3], m[2], or m[1].

m[4] will always be the datsource; m[5] will be the timestamp.

this change adds unit tests to demonstrate the expected parseTaskId
output for various tasks type supported. it also covers the error case
when taskId doesn't match the regular expression.

alperkokmen commented:
@clintropolis i found out `parseTaskId` was broken in an earlier commit while adding unit tests. 117cc36 addresses that regression with unit tests to make sure it works the way it was intended to for known tasks. ea4f4db adds the support for not throwing an error for not-so-commonly-used tasks.

clintropolis reviewed (approved):
Awesome thanks! LGTM :+1:

dclim commented:
👍
